apps/server/src/shared/domain/value-object.spec.ts

```diff
@@ -0,0 +1,187 @@
+		});
+
+		it('should be return true by different references', () => {
+			const { nameThor1, nameThor2 } = setup();
+
+			expect(nameThor1.equals(nameThor2)).toBe(true);
+		});
+	});
+
+	describe('By usage of primitive array as value object', () => {
+		const setup = () => {
+			class Names extends ValueObject<string[]> {}
+
+			const nameThors1 = new Names(['Thor1', 'Thor2', 'Thor3']);
+			const nameThors2 = new Names(['Thor1', 'Thor2', 'Thor3']);
+			const nameThorsDifferent = new Names(['Thor1', 'Thor2', 'Thor4']);
+			const nameThorsDifferentCount = new Names(['Thor1', 'Thor2']);
+
+			return { nameThors1, nameThors2, nameThorsDifferent, nameThorsDifferentCount };
+		};
+
+		it('should be return true by same reference', () => {
+			const { nameThors1 } = setup();
+
+			expect(nameThors1.equals(nameThors1)).toBe(true);
+		});
+
+		it('should be return true by different references', () => {
+			const { nameThors1, nameThors2 } = setup();
+
+			expect(nameThors1.equals(nameThors2)).toBe(true);
+		});
+
+		it('should be return false by different values inside the array', () => {
+			const { nameThors1, nameThorsDifferentCount } = setup();
+
+			expect(nameThors1.equals(nameThorsDifferentCount)).toBe(false);
+		});
+
+		it('should be return false by different values inside the array', () => {
+			const { nameThors1, nameThorsDifferent } = setup();
+
+			expect(nameThors1.equals(nameThorsDifferent)).toBe(false);
+		});
+	});
+
+	describe('When value object with validation is created.', () => {
+		const setup = () => {
+			class Name extends ValueObject<string> {
+				protected validation(value: unknown): boolean {
+					let isValid = false;
+
+					if (TypeGuard.isString(value) && this.isValidLength(value)) {
+						isValid = true;
+					}
+
+					return isValid;
+				}
+
+				private isValidLength(value: string): boolean {
+					let isValid = false;
+
+					if (value.length > 0 && value.length <= 5) {
+						isValid = true;
+					}
+
+					return isValid;
+				}
+			}
+
+			return { Name };
+		};
+
+		it('validation should be execute and throw if not valid', () => {
+			const { Name } = setup();
+
+			const expectedError = new Error('ValueObject Name validation is failed for input ""');
+			expect(() => new Name('')).toThrowError(expectedError);
+		});
+
+		it('should execute but not throw if is valid value is passsed', () => {
+			const { Name } = setup();
+
+			expect(new Name('Thor')).toBeDefined();
+		});
+		/*
+		it('should execute but not throw if is valid value is passsed', () => {
+			const { Name } = setup();
+
+			const myName = new Name('Thor');
+
+			expect(myName.isValidValue('123456')).toBe(false);
+			expect(myName.isValidValue('')).toBe(false);
+			expect(myName.isValidValue('ABC')).toBe(true);
+		});
+		*/
+	});
+
+	describe('When value object with modification is created.', () => {
+		const setup = () => {
+			class Name extends ValueObject<string> {
+				protected modified(value: string): string {
```

Comment:
Replacing the boilerplate with array of functions that can be execute will improve the usability 
modifications: [this.removeWithspacesAtBeginning, this.truncatedToLength4]
validations: [TypeGuard.isString, this.isValidLength]

In this case it is also easy to use same hooks for different value objects. But it is harder to implement with ts. We should look togehter into it.